pymc3/step_methods/hmc/trajectory.py

```diff
@@ -57,6 +57,13 @@ def _theano_energy_function(H, q, **theano_kwargs):
     return energy_function, p
 
 
+def _theano_velocity_function(H, p, **theano_kwargs):
+    v = H.pot.velocity(p)
```

Comment:
What I am getting at is that wherever you call `self.compute_velocity`, you can call `H.pot.velocity`.  I dug some more and `H.pot.velocity` calls to one of the functions in `quadpotential.py`, and is multiplying the momentum by a scaling constant or matrix.  This should be pretty performant, but I haven't tested whether compiling to theano makes it faster.  Just considering surface area of code, I'd rather not add another function though.

Separately! We pass the gradient through since that's a fairly expensive operation.  Shouldn't velocity be resampled at the start of each leapfrog step, so returning it doesn't help things?  I flagged another comment on the one place where it might be useful, but I don't know why it is useful there!